### PR TITLE
Add promptflow-runtime for curated environment

### PR DESCRIPTION
**Pull request by zhengfeiwang**
This PR targets to add `promptflow-runtime` for curated environment.

### CODE REVIEW

**Review by wangchao1230 — DISMISSED** [review]

File: pipelines/environments/promptflow-runtime/spec.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+$schema: https://azuremlschemas.azureedge.net/latest/environment.schema.json
+
+description: >-
+  Runtime environment for PromptFlow.
```
Comment (wangchao1230) [inline pipelines/environments/promptflow-runtime/spec.yaml:4]: we need to add an aka.ms link to changelog here.

Comment (zhengfeiwang) [reply pipelines/environments/promptflow-runtime/spec.yaml]: updated with CHANGELOG link

**Review by wangchao1230 — APPROVED** [review]

File: pipelines/environments/promptflow-runtime/context/Dockerfile [hunk 2]
```diff
@@ -0,0 +1 @@
+FROM mcr.microsoft.com/azureml/promptflow/promptflow-runtime:{{latest-image-tag}}
```
Comment (elesel) [inline pipelines/environments/promptflow-runtime/context/Dockerfile:1]: We don't create registry environments just to point to other images. Please include the actual Docker build context of this environment here.

**Review by elesel — CHANGES_REQUESTED** [review]

File: pipelines/environments/promptflow-runtime/context/Dockerfile (hunk 2 above)
Comment (vizhur) [reply pipelines/environments/promptflow-runtime/context/Dockerfile:1]: base image has out of SLA vulnerability and doesn't follow convention for tagging

"VulnerabilityId": 992484,
"AssetId": sha256:778dc5423f681247d485658271196c44d976d978cb52fe78d077302196ef428a,
"RunId": 2023-05-16T13:51:22.3306902Z,
"ImageName": azuremlmcr.azurecr.io/public/azureml/promptflow/promptflow-runtime:latest,
"IsActionable": true,
"VulnerabilityName": GO (Go) Security Update for google.golang.org/protobuf (GHSA-hw7c-3rfg-p46j),
"DueDate": 2023-04-15T00:00:00Z,

Comment (zhengfeiwang) [reply pipelines/environments/promptflow-runtime/context/Dockerfile:1]: For the tagging issue, we can update our pipeline to not push the latest image to `azuremlmcr`.

And for the golang issue, we will try to fix that.

Comment (zhengfeiwang) [reply pipelines/environments/promptflow-runtime/context/Dockerfile:1]: have updated to fix the golang vulnerability issue, and the image tag is pushed with both YYYYMMDD.vX and latest.

Comment (zhengfeiwang) [reply pipelines/environments/promptflow-runtime/context/Dockerfile:1]: and for the issue Louie pointed out, let's treat it as a short-term solution, will include related build context after the PuP of related SDK - resolving this discussion.

**Review by wangchao1230 — APPROVED** [review]

File: pipelines/environments/promptflow-runtime/spec.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+build:
+  path: "{{asset.repo.url}}#{{asset.repo.commit_hash}}:{{asset.repo.build_context.path}}"
+  dockerfile_path: "{{image.dockerfile.path}}"
+
+os_type: linux
+
+tags:
+  OS: Ubuntu20.04
+  Python: 3.8
```
Comment (wangchao1230) [inline pipelines/environments/promptflow-runtime/spec.yaml:17]: add a preview tag?

Comment (zhengfeiwang) [reply pipelines/environments/promptflow-runtime/spec.yaml:17]: added

**Review by wangchao1230 — APPROVED** [review]

File: pipelines/environments/promptflow-runtime/asset.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+name: promptflow-runtime
+version: auto
+type: environment
+spec: spec.yaml
+extra_config: environment.yaml
+categories: ["PromptFlow"]
```
Comment (elesel) [inline pipelines/environments/promptflow-runtime/asset.yaml:6]: Which is correct, "PromptFlow" or "prompt flow"? See #646

Comment (zhengfeiwang) [reply pipelines/environments/promptflow-runtime/asset.yaml]: updated here to follow the "prompt flow"

**Review by elesel — APPROVED** [review]

**Pull request merged**